spec/lib/chess/core/record_spec.rb

```diff
@@ -0,0 +1,157 @@
+require 'spec_helper'
+require 'chess/core/record'
+require 'chess/core/board'
+
+require_relative '../../../helpers/record_pieces'
+
+Intention = Struct.new(:type, :origin_cell, :target_cell)
+
+RSpec.describe Chess::Core::Record do
+  include Helpers::RecordPieces
+
+  subject(:record) { described_class.new }
+
+  let(:board) { Chess::Core::Board.new }
+
+  before do
+    board.generate_cells
+    hardcode_record_case1(board)
+  end
+
+  describe '#add' do
+    describe 'when intention is type :move and no capture' do
+      let(:accomplished_intention) { Intention.new(:move, board.cells[0][4], board.cells[1][4]) }
+
+      it 'returns a Chess::Core::Record::CommitRecord' do
+        expect(
+          record.add(accomplished_intention, Chess::WHITE_TEAM, nil)
```

Comment:
Use `let` to store the result of this in a helper variable and call it here, just for more clarity. Also you could reuse it on all specs below

Reply:
done!